core/state/access_witness.go

```diff
@@ -244,7 +244,7 @@ type branchAccessKey struct {
 
 func newBranchAccessKey(addr []byte, treeIndex uint256.Int) branchAccessKey {
 	var sk branchAccessKey
-	copy(sk.addr[:], addr)
+	copy(sk.addr[20-len(addr):], addr)
```

Comment:
If `len(addr)<20` this was doing the copy incorrectly.

Another option for this particular bug case is **not** doing this code line change, and do:
![image](https://github.com/gballet/go-ethereum/assets/6136245/28013358-38eb-4068-b18f-615a2dd5ca54)

But I think this proposed change is more general, and if we have a similar situation in other place the current fix would cover it.

Open to opinions though.